app/controllers/admin/administrators_controller.rb

```diff
@@ -0,0 +1,32 @@
+class Admin::AdministratorsController < Admin::BaseController
+  load_and_authorize_resource
+
+  def index
+    @administrators = @administrators.page(params[:page])
+  end
+
+  def search
+    @user = User.find_by(email: params[:email])
+
+    respond_to do |format|
+      if @user
+        @administrator = Administrator.find_or_initialize_by(user: @user)
+        format.js
+      else
+        format.js { render "user_not_found" }
+      end
+    end
+  end
+
+  def create
+    @administrator.user_id = params[:user_id]
+    @administrator.save
+
+    redirect_to admin_administrators_path
+  end
+
+  def destroy
```

Comment:
Okey added that scenario to the spec https://github.com/consul/consul/pull/1560/commits/e9ea72113b30968e46225315e5e42ac97caea46b and simple current user check on the controller https://github.com/consul/consul/pull/1560/commits/fe5ddd019bc66fc95dffa8233b31b4552bcdf5a6

Now if the current user tries to remove himself from the administrators list he will get:
![screen shot 2017-05-25 at 11 31 08](https://cloud.githubusercontent.com/assets/983242/26444706/b91cc890-413d-11e7-8efb-86b085d4433c.jpg)

Any more details that could be polished? :)